<a href="https://colab.research.google.com/github/dannabarrera292002PROGRAMACION/PROGRAMACION-/blob/main/taller_portafolio_inversion.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 📊 Taller Práctico: Análisis de un Portafolio de Inversión
## Lenguajes de Programación – Ingeniería Financiera

En este taller practicarás el uso de **tuplas, listas y diccionarios** para analizar un portafolio de inversión.

### Objetivos
- Usar **tuplas** para información fija
- Usar **listas** para almacenar datos
- Usar **diccionarios** para relacionar información
- Practicar **indexación, append() y len()**
- Analizar un portafolio de inversión



## Parte 1 — Tuplas (Información fija)
Las tuplas son útiles cuando los datos **no cambian**.

In [2]:
empresas = ("Apple", "Microsoft", "Amazon", "Tesla", "Google")
print(empresas)

('Apple', 'Microsoft', 'Amazon', 'Tesla', 'Google')


### Ejercicio
Imprime:
- La primera empresa
- La tercera empresa
- La última empresa

In [4]:
# Escribe tu código aquí
print(empresas[0])
print(empresas[2])
print(empresas[4])

Apple
Amazon
Google


## Parte 2 — Lista (Precio de las acciones)

In [6]:
precios_acciones = [185, 320, 145, 250, 275]
print(precios_acciones)

[185, 320, 145, 250, 275]


### Ejercicio
Imprime:
- Precio de Apple
- Precio de Tesla
- Cantidad de empresas en el portafolio usando len()

In [8]:
# Escribe tu código aquí
print(precios_acciones[0])
print(precios_acciones[3])
print(len(empresas))

185
250
5


## Parte 3 — Diccionario (Cantidad de acciones)

In [10]:
acciones_portafolio = {
    "Apple": 10,
    "Microsoft": 5,
    "Amazon": 8,
    "Tesla": 4,
    "Google": 6
}
print(acciones_portafolio)

{'Apple': 10, 'Microsoft': 5, 'Amazon': 8, 'Tesla': 4, 'Google': 6}


### Ejercicio
Imprime:
- Acciones de Apple
- Acciones de Google

In [11]:
# Escribe tu código aquí
print(acciones_portafolio["Apple"])
print(acciones_portafolio["Google"])

10
6


## Parte 4 — Valor invertido

In [21]:
valor_apple = precios_acciones[0] * acciones_portafolio["Apple"]
print(valor_apple)

1850


### Ejercicio
Calcula el valor invertido en Microsoft, Amazon, Tesla y Google.

In [34]:
# Escribe tu código aqui
valor_microsoft = precios_acciones[1] * acciones_portafolio["Microsoft"]
valor_amazon = precios_acciones[2] * acciones_portafolio["Amazon"]
valor_tesla = precios_acciones[3] * acciones_portafolio["Tesla"]
valor_google = precios_acciones[4] * acciones_portafolio["Google"]

print ("el valor total es:", valor_microsoft)
print(valor_amazon)
print(valor_tesla)
print(valor_google)

print((valor_tesla * 10/100) + valor_tesla)

el valor total es: 1600
1160
1000
1650
1100.0


## Parte 5 — Lista del portafolio

In [26]:
valor_portafolio = []
valor_portafolio.append(valor_apple)
valor_portafolio.append(valor_microsoft)
valor_portafolio.append(valor_amazon)
valor_portafolio.append(valor_tesla)
valor_portafolio.append(valor_google)

print(valor_portafolio)
# Agrega los demás valores calculados

[1850, 1600, 1160, 1000, 1650]


In [35]:
print("Valor total del portafolio:", sum(valor_portafolio))
print("Promedio:", sum(valor_portafolio)/len(valor_portafolio))
print("Número de empresas:", len(valor_portafolio))

Valor total del portafolio: 7260
Promedio: 1452.0
Número de empresas: 5


## Reflexión
1. ¿En qué empresa hay mayor inversión? Apple 1850
2. ¿El portafolio está diversificado?
3. ¿Qué pasaría si Tesla sube 10%? El valor subiría a 1100
4. ¿Por qué son útiles las estructuras de datos en finanzas?
Para ordenar datos en un espacio y poder acceder a ellos facilmente